pkg/internal/kube/informer.go

```diff
@@ -262,11 +263,27 @@ func LoadConfig(kubeConfigPath string) (*rest.Config, error) {
 	return config, nil
 }
 
-func (k *Metadata) initInformers(ctx context.Context, client kubernetes.Interface, syncTimeout time.Duration) error {
+func (k *Metadata) initInformers(ctx context.Context, client kubernetes.Interface, restrictNode string, syncTimeout time.Duration) error {
 	if syncTimeout <= 0 {
 		syncTimeout = defaultSyncTimeout
 	}
-	informerFactory := informers.NewSharedInformerFactory(client, resyncTime)
+	var informerFactory informers.SharedInformerFactory
+	if restrictNode == "" {
+		k.log.Info("no node selector provided. Listening to global resources")
+		informerFactory = informers.NewSharedInformerFactory(client, resyncTime)
+	} else {
+		fieldSelector := fields.OneTermEqualSelector("spec.nodeName", restrictNode).String()
+		k.log.Info("using field selector", "selector", fieldSelector)
```

Comment:
do we want to keep that as info? or debug would be better?

Reply:
you are right. Moving to debug would have more sense